### PR TITLE
Update and restructure AMD documentation

### PR DESCRIPTION
**Pull request by JohanWBorg**

### CODE REVIEW

**Review by stellabelin — CHANGES_REQUESTED** [review]
I think it's great that the pages are split up, I think it will be clearer. General thoughts:

1. "parts" is used a lot to describe the different steps from strategy but I find it a little confusing. I think sentences like "When running the residual part" can be replaced with something like "When developing the residual error model" or "When running the residual error subtool". Titles like "Strategy parts" is similarly a bit confusing to me, maybe "Strategy components" or "Strategy - Model building components" (which is perhaps too wordy)
2. I think it would be useful to define what the different types of covariates mean on the AMD main page. The separate model type pages can then just have the default search space if needed.
3. Covsearch is sometimes written as "covsearch" and sometimes as "COVSearch", this should be consistent
4. Some tables are formatted differently (e.g. in drug metabolite on lines 126-138), I think what we have in most places is code formatting on the argument names
5. For the code examples, they have different formatting when it comes to tabbing etc, this should be consistent as well
6. Examples now have to include relevant initial estimates, they have no real default anymore. The text reflects this but the runs at the beginning of the page does not

Overall spelling things: 

1. "modeltype" -> "model type"
2. "searchspace" -> "search space"

File: docs/amd_drug_metabolite.rst [hunk 1]
```diff
@@ -0,0 +1,421 @@
+.. _amd_drug_metabolite:
+
+=====================
+AMD - Drug metabolite
+=====================
+
+Will create a drug metabolite model based on a model or dataset input. The given workflow is based on the given :ref:`strategy<strategy_amd>`.
+
+Regarding the dataset, two DVIDs are expected.
```
[the remark this answers is not in the document]
Comment (JohanWBorg) [reply docs/amd_drug_metabolite.rst:9]: No there are no checks on top level (maybe this is something we want?) but it will fail when running structsearch as adding a metabolite compartment assumes two DVIDs. Will change it to required nonetheless

File: docs/amd_drug_metabolite.rst [hunk 2]
```diff
@@ -0,0 +1,421 @@
...
+~~~~~~~
+
+The code to initiate the AMD tool for a PK model:
+
+.. pharmpy-code::
+
+    from pharmpy.tools import run_amd
+
+    dataset_path = 'path/to/dataset'
+    strategy = 'default'
```
[the remark this answers is not in the document]
Comment (JohanWBorg) [reply docs/amd_drug_metabolite.rst:33]: moved it to within the run statement

File: docs/amd_drug_metabolite.rst [hunk 3]
```diff
@@ -0,0 +1,421 @@
...
+
+
+**Structural covariates**
+
+The structural covariates are added directly to the starting model. If these cannot be added here (due to missing 
+parameters for instance) they will be tested once more at the start of the next covsearch run.
+
+Note that all structural covariates are added all at once without any test or search.
+
+These are given within the search space by specifying them as mechanistic covariates in the following way:
```
[the remark this answers is not in the document]
Comment (JohanWBorg) [reply docs/amd_drug_metabolite.rst:105]: My mistake. Accidentally wrote mechanistic covariates instead but changed it to be a bit clearer
"The structural covariates are given within the search space by specifying them in the following way:"

File: docs/amd_drug_metabolite.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (stellabelin) [reply docs/amd_drug_metabolite.rst:9]: If it fails I think it should be caught at top level (not in this PR though).

[the remark this answers is not in the document]
Comment (JohanWBorg) [reply docs/amd_drug_metabolite.rst:9]: Of course, I'll open an issue for this

File: docs/amd_drug_metabolite.rst [hunk 4]
```diff
@@ -0,0 +1,421 @@
...
+**Exploratory covariates**
+
+The covariate effects remaining in the search space after having run potentially both structural and mechanistic covariates
+are now run in an exploratory search.
+
+**Examples**
+
+.. code-block::
+
+    mechanistic_covariates = [AGE, (CL,WGT)]
```
[the remark this answers is not in the document]
Comment (JohanWBorg) [reply docs/amd_drug_metabolite.rst:359]: Yes it is an AMD argument. But the argument need to be specified in the example as well (otherwise i cannot show how it affects the search space. but i can move it outside of the code block

File: docs/amd.rst [hunk 5]
```diff
@@ -41,41 +61,57 @@ Arguments
...
 +---------------------------------------------------+-----------------------------------------------------------------------------------------------------------------+
-| ``met_init``                                      | Initial estimate for the mean equilibration time (only for pkpd models, default is 0.1)                         |
+| ``dv_types``                                      | Dictionary of DV types for multiple DVs (e.g. dv_types = {'target': 2}). Default is None.                       |
+|                                                   | Allowed keys are: 'drug', 'target', 'complex', 'drug_tot' and 'target_tot'. (For TMDD models only)              |
+|                                                   | For more information see :ref:`here<dv_types>`.                                                                 |
++---------------------------------------------------+-----------------------------------------------------------------------------------------------------------------+
+
+
+Common arguments
```
[the remark this answers is not in the document]
Comment (JohanWBorg) [reply docs/amd.rst:98]: Used "General" instead

File: docs/amd.rst [hunk 6]
```diff
@@ -41,41 +61,57 @@ Arguments
...
-| ``ec50_init``                                     | Initial estimate for the EC50 (only for pkpd models, default is 0.1)                                            |
+| ``met_init``                                      | Initial estimate for the mean equilibration time (only for pkpd models)                                         |
 +---------------------------------------------------+-----------------------------------------------------------------------------------------------------------------+
-| ``met_init``                                      | Initial estimate for the mean equilibration time (only for pkpd models, default is 0.1)                         |
+| ``dv_types``                                      | Dictionary of DV types for multiple DVs (e.g. dv_types = {'target': 2}). Default is None.                       |
+|                                                   | Allowed keys are: 'drug', 'target', 'complex', 'drug_tot' and 'target_tot'. (For TMDD models only)              |
+|                                                   | For more information see :ref:`here<dv_types>`.                                                                 |
```
[the remark this answers is not in the document]
Comment (JohanWBorg) [reply docs/amd.rst:94]: Can't the same logic be applied to the arguments `b_init`, `emax_init`, `ec50_init`, and `met_init` which are only applied to PKPD models?
I see your point in moving the arguments to their respective page but it feels weird to only remove some of the arguments in my opinion. Perhaps the "model type specific arguments" should only be located on their respective page instead

[the remark this answers is not in the document]
Comment (stellabelin) [reply docs/amd.rst:94]: Sorry, I was a bit unclear! I meant that the argument itself can maybe be kept here, but on the TMDD page for example it is written which are allowed. Was thinking if we have more model types it will be pretty dense to have allowed keys for each type on the main page.

[the remark this answers is not in the document]
Comment (JohanWBorg) [reply docs/amd.rst:94]: Aaaah! Ok! Absolutely, I can reduce the description the general page!

File: docs/amd_drug_metabolite.rst [hunk 7]
```diff
@@ -0,0 +1,421 @@
...
+            s1 [label="modelsearch"]
+            s2 [label="structsearch"]
+
+            base -> s0
+            s0 -> s1
+            s1 -> s2
+        }
+
+
+**Structural covariates**
```
[the remark this answers is not in the document]
Comment (JohanWBorg) [reply docs/amd_drug_metabolite.rst:98]: Created a description on the general page instead and moved the example(s) and information there as it is shared between the different model types

**Review by stellabelin — APPROVED** [review]
LGTM!
